llvm/test/MC/AArch64/SVE2/directive-arch_extension-negative.s

```diff
@@ -6,10 +6,10 @@ tbx z0.b, z1.b, z2.b
 // CHECK: error: instruction requires: sve2 or sme
 // CHECK-NEXT: tbx z0.b, z1.b, z2.b
 
-.arch_extension sve2-aes
-.arch_extension nosve2-aes
+.arch_extension sve-aes
+.arch_extension nosve-aes
```

Comment:
Here you are replacing the old test instead of adding a new one. Now the (no)sve2-aes directives are not tested anywhere. Similarly in other files.

Reply:
That's because all `sve2-aes` does is enable `sve2` and `sve-aes` (which these instructions are guarded by). The following sequence:
```
.arch_extension sve2-aes
.arch_extension nosve2-aes
```

Would still allow the use of `aesd`, because disabling a specific feature does not disable it's dependencies. You raise a good point with this comment, we may need to reconsider if this is the intended behavior.

In either case, I should not have removed the positive tests for `sve2-aes` so I'll fix that.

Reply:
That's because all `sve2-aes` does is enable `sve2` and `sve-aes` (which these instructions are guarded by). The following sequence:
```
.arch_extension sve2-aes
.arch_extension nosve2-aes
```

Would still allow the use of `aesd`, because disabling a specific feature does not disable it's dependencies. You raise a good point with this comment, we may need to reconsider if this is the intended behavior.

In either case, I should not have removed the positive tests for sve2-aes so I'll fix that.

Reply:
> all sve2-aes does is enable sve2 and sve-aes

Correct.

> disabling a specific feature does not disable it's dependencies

Also correct

To preserve the old semantics of `nosve2-aes` we need to change the asm parser as follows:
`{"sve2-aes", {AArch64::FeatureSVE2, AArch64::FeatureSVEAES}}`

Reply:
I see, thanks.

Reply:
You may find in useful to look at https://github.com/llvm/llvm-project/pull/110816/files. In order to make nosve2-aes backwards compatible on the command line you need to adjust AArch64::ExtensionSet::disable in TargetParser. Then you can add tests in clang/test/Driver/ for it.